server/controllers/article.controller.js

```diff
@@ -204,5 +209,92 @@ class ArticleController {
     await sendMail(mailOption);
     return Response(res, 200, 'Article shared successfully');
   }
+
+  /**
+    * @description gets a single article
+    * @param {object} req
+    * @param {object} res
+    * @returns {object} An article and it's associated comments
+    */
+  static async getOneArticle(req, res) {
+    const { slug: articleSlug } = req.params;
+    const existingArticle = await Article.findOne({
+      where: { slug: articleSlug },
+      include: [
+        {
+          model: User,
+          as: 'author',
+          attributes: ['id', 'username', 'bio', 'image']
+        }
+      ],
+    });
+
+    if (!existingArticle) return Response(res, 404, 'Article not found.');
+
+    const {
+      dataValues: {
+        id, slug, title, description, body, tags, author, published
+      }
+    } = existingArticle;
+
+    const commentsTable = published ? Comment : DraftComment;
+    const articleProperties = await Promise.all([
+      Clap.count({ where: { articleId: id } }),
+      commentsTable.findAndCountAll({
+        where: {
+          articleId: id,
+          parentId: null
+        }
+      }),
+      commentsTable.count({ where: { articleId: id } })
+    ]);
+
+    const articleObject = {
+      id,
+      slug,
+      title,
+      description,
+      body,
+      tags,
+      author,
+      claps: `${articleProperties[0]} claps`,
+    };
+
+    const { authorization } = req.headers;
+    if (authorization) {
+      const token = authorization.split(' ')[1];
+      const decoded = await verifyToken(token);
+      await Readstat.findOrCreate({
+        where: { userId: decoded.id },
+        defaults: {
+          userId: decoded.id,
+          articleId: id,
+          slug,
+        },
+      });
+    }
+
+    const mainComments = articleProperties[1].rows;
+    if (mainComments.length) {
+      const commentCount = articleProperties[2];
+      let counter = 0;
+      mainComments.forEach(async (comment) => {
+        const threadComment = await commentsTable.findAll({
+          where: { parentId: comment.dataValues.id }
+        });
+        comment.dataValues.threadcomments = threadComment;
+        counter += 1;
+
+        if (counter === mainComments.length) {
+          articleObject.comments = mainComments;
+          articleObject.commentCount = commentCount;
+          return Response(res, 200, 'Article found.', [articleObject]);
+        }
+      });
+    } else {
+      articleObject.comments = '0 comments';
+      return Response(res, 200, 'Article found.', [articleObject]);
```

Comment:
Hi @TheDrizzyWay , looks like you did not call the getArticleRatings method here. The article's rating should be passed in the response. This is so that every article being returned will have its rating i

Reply:
Thanks for pointing that out. Will include it right away.